pkg/system/lstat_windows.go

```diff
@@ -12,3 +26,223 @@ func Lstat(path string) (*StatT, error) {
 
 	return fromStatT(&fi)
 }
+
+func (fi *fileInfo) Size() int64 {
+	return fi.data.Size()
+}
+
+func (fi *fileInfo) Mode() (m os.FileMode) {
+	m = fi.data.Mode()
+
+	if fi.maskReparse {
+		m &^= os.ModeSymlink
+	}
+
+	return m
+}
+
+func (fi *fileInfo) ModTime() time.Time {
+	return fi.data.ModTime()
+}
+
+// Sys returns syscall.Win32FileAttributeData for file fs.
+func (fi *fileInfo) Sys() interface{} {
+	return fi.data.Sys()
+}
+
+func shouldMaskReparsePoint(name string, namep *uint16) (bool, error) {
+	var fd syscall.Win32finddata
+
+	h, err := syscall.FindFirstFile(namep, &fd)
+	if err != nil {
+		return false, &os.PathError{"FindFirstFile", name, err}
+	}
+	syscall.FindClose(h)
+
+	fullpath := name
+	if !isAbs(fullpath) {
+		fullpath, err = syscall.FullPath(fullpath)
+		if err != nil {
+			return false, &os.PathError{"FullPath", fullpath, err}
+		}
+	}
+
+	if fd.FileAttributes&syscall.FILE_ATTRIBUTE_REPARSE_POINT != 0 {
+		if fd.Reserved0&reparsePointNameSurrogate == 0 {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func isAbs(path string) (b bool) {
```

Comment:
Maybe it makes sense to mention the below is a fork of (copy-paste from) Go's `src/os/path_windows.go`, so that if they fix something in there we can also update it?